src/util/prop-definition.js

```diff
@@ -54,7 +57,13 @@ export default class PropDefinition {
       // Only accept documented options and perform minimal input validation.
       switch (option) {
         case 'attribute':
-          this.attrName = resolveAttrName(optVal, nameOrSymbol);
+          if (isObject(optVal)) {
+            this.attrSource = resolveAttrName(optVal.source, nameOrSymbol);
+            this.attrTarget = resolveAttrName(optVal.target, nameOrSymbol);
+          } else {
+            this.attrSource = resolveAttrName(optVal, nameOrSymbol);
+            this.attrTarget = this.attrSource;
```

Comment:
Wherever we can it would be nice to save references to the props we're accessing. Sort of a nit, though.